en/setup/vscode.md

```diff
@@ -0,0 +1,83 @@
+
+## Installation & Setup
+
+1. [Download and install VSCode](https://code.visualstudio.com/) (you will be offered the correct version for your OS).
+1. Open VSCode and add the PX4 source code: 
+   - Select *Open folder ...* option on the welcome page (or using the menu: **File > Open Folder**):
+     ![Open Folder](../../assets/vscode/welcome_open_folder.jpg)
+   - A file selection dialog will appear.
+     Select the PX4 **Firmware** directory and then press **OK**.
+
+   The project files and configuration will then load into *VSCode*.
+1. Press **Install All** on the *This workspace has extension recommendations* prompt (this will appear on the bottom right of the IDE).
+   ![Install extensions](../../assets/vscode/prompt_install_extensions.jpg)
+
+   VSCode will open the Extensions panel on the left hand side so you can watch the progress of installation.
+
+   ![PX4 loaded into VSCode Explorer](../../assets/vscode/installing_extensions.jpg)
+1. A number of notifications/prompts may appear in the bottom right corner
+   > **Tip** If the prompts disappear, click the little "alarm" icon on the right of the bottom blue bar.
+
+   - If prompted to install a new version of *cmake*: 
+     - Say **No** (the right version is installed with the [PX4 developer environment](../setup/dev_env.md)).
+   - If prompted to sign into *github.com* and add your credentials:
+     - This is up to you! It provides a deep integration between Github and the IDE, which may simplify your workflow.
+   - Other prompts are optional, and may be installed if they seem useful. <!-- perhaps add screenshot of these prompts -->
+
+
+## Building PX4 {#building}
+
+To build:
+1. Select your build target ("cmake build config"):
+   - The current *cmake build target* is shown on the blue *config* bar at the bottom (if already your target, skip to next step).
+     ![Select Cmake build target](../../assets/vscode/cmake_build_config.jpg)
+   - Click the target on the config bar to display other options, and select the one you want (this will replace any selected target).
+   - *Cmake* will then configure your project (see notification in bottom right).
+     ![Cmake config project](../../assets/vscode/cmake_configuring_project.jpg).
+   - Wait until this completes.
+     When this is done the notification will disappear and you'll be shown the build location:
+     ![Cmake config project](../../assets/vscode/cmake_configuring_project_done.jpg).
+1. You can then kick off a build from the config bar (select either **Build** or **Debug**).
+   ![Run debug or build](../../assets/vscode/run_debug_build.jpg)
+
+After building at least once you can now use [code completion](#code completion) and other *VSCode* features.
+
+<!-- px4 at the bottom - select target to launch - test mixer multi_rotor or PX4 -->
+
+## Debugging PX4 {#debugging_sitl}
+
```

Comment:
@dagar This is the bit that gives me trouble - I don't know how to select and executable and make it launch SITL. I would also like to be able to make the launch headless.

![image](https://user-images.githubusercontent.com/5368500/65144322-77015a80-da5a-11e9-8fe4-6280c5cd0b00.png)